code/modules/mob/living/bot/secbot.dm

```diff
@@ -521,8 +523,9 @@
 
 /mob/living/bot/secbot/attack_hand(mob/living/carbon/human/M as mob)
 	..()
-
-	if(M.a_intent == I_HURT ) //assume he wants to hurt us.
+	if(!on)
+		return
+	if((M.a_intent == I_HURT) && on ) //assume he wants to hurt us.
```

Comment:
The check above already checks it
```suggestion
	if((M.a_intent == I_HURT)) //assume he wants to hurt us.
```

Reply:
I was hoping you'd notice that.
I wanted to draw attention to it because I don't know what's causing securitrons to still remember that Charlie McBobbarty touched them with the harm intent even if they're offline.

Reply:
It shouldn't. If it does, that means that Charlie McBobbarty was a target from the past. At that point you need to add code inside of think() state machine which gets rid of target when we are turned off

Reply:
Just to clarify: when I check with the SecHUD, it still broadcasts its empty-hand attack message as soon as it gets switched on, even though the "attack" was made while it was off. Adding the conditional check to see if it's on prevents the message from broadcasting WHILE it was switched off (which is how it is now on the current build), but it then gets sent when switched on again and the target is set to the attacker. Only then does the *Arrest* status also gets set.
I'll try nullifying the target though in the think() proc when it checks if it is on or not.

Reply:
Doesn't seem to have done it.
I must have missed something or misunderstood something you had in mind.

I'm noting though, that when switched back on again, it broadcasts the message twice. This was also before the attempted fix just now.

Reply:
Hitting Beepsky with a wrench while offline doesn't do anything, and turning him back on doesn't cause him to attack you.
But, hitting Beepsky with a wrench while offline and then IMMEDIATELY turning him on gives a generic attack message (not the same message if you were to attack him while online, which would indicate what he was attacked with).

Reply:
I'm also noting that there's two attack_hand and attackby proc definitions but both are different. I'm not sure if that's deliberate though I am leaning towards 'it might not be', but I didn't want to touch anything there just in case.

Reply:
I see, that is a major problem. One of them overrides the other, which is my mistake as I initially added the second definition.

Solution:

- Delete proc entirely that is on line 159-167

- rename proc on line 101 to something like `open_interface(var/mob/user)`. Then add call of the proc `open_interface(M)` after line 533 but outside of if check(meaning it is on same indentation as `if(M.a_intent == I_HURT)`).

Reply:
I did as you suggested.
Attacking a securitron with a tool while it's offline means it never reports the attack -- the desirable behaviour, so I suppose we can say that's resolved.

However, I tried making an open_interface proc and I added it on line 540 (lines 533 to 539 seemed to still be part of the same IF check) but that just meant the user interface couldn't be opened if it was switched off. The issue still persisted as well.
So I resorted to adding it on line 522, just before the if(!on) check. It didn't seem to resolve the issue either, unfortunately.

Reply:
Did you rename proc Into open_interface or did you make new?

Reply:
Both.
I renamed the first attack_hand definition to open_interface but had to make it a new proc because it's not defined anywhere else.